File: Problems/1582. Special Positions in a Binary Matrix/1582. Special Positions in a Binary Matrix.java
import java.util.ArrayList;
import java.util.List;
class Solution {
    public int numSpecial(int[][] mat) {
        int m = mat.length;
        int n = mat[0].length;
        int res = 0; // record the result.
        List<Integer> colCand = new ArrayList<>();

        // Find all column candidate.
        for (int[] row: mat) {
            int nOne = 0;
            int pos = 0;
            for (int j = 0; j < n; j++) {
                if (row[j] == 1) {
                    nOne += 1;
                    pos = j; 
                }
            }
            if (nOne == 1) colCand.add(pos);
        }

        // Verify candidate columns.
        for (int col: colCand) {
            int nOne = 0;
            for (int i = 0; i < m; i++)
                nOne += mat[i][col];
            if (nOne == 1) res += 1;
        }

        return res;
    }
}

File: Problems/1582. Special Positions in a Binary Matrix/1582. Special Positions in a Binary Matrix.py
from typing import List;
from collections import defaultdict;
class Solution:
    def numSpecial(self, mat: List[List[int]]) -> int:
        # Count size of the matrix.
        m = len(mat);
        n = len(mat[0]);
        # Find candidates through row and column respectively.
        # Find the index that is the only "1" in that row/column.
        # Then return the size of intersection of sets of candidates.
        colCand = set();
        rowCand = set();
        
        # Find all row-special numbers.
        for i in range(m):
            pos = -1;
            numOne = 0;
            for j in range(n):
                if mat[i][j] == 1:
                    pos = j;
                    numOne += 1;
            if numOne == 1:
                rowCand.add((i, pos));
        
        # Find all col-special number.
        for j in range(n):
            pos = -1;
            numOne = 0;
            for i in range(m):
                if mat[i][j] == 1:
                    pos = i;
                    numOne += 1;
            if numOne == 1:
                colCand.add((pos, j));
        
        # Return the intersection.
        return len(colCand.intersection(rowCand));